### PR TITLE
The Servant rework (Maid/Butler QOL)

### PR DESCRIPTION
**Pull request by AceVX**
_"Djedor? Oh, he is our butler, But really he's a member of the family. A loyal and reliable servant for many, many years. Trusted to keep the house, manor grounds, kitchen and lesser servants in working order. He cooks, serves wine, The old man even taught the children music and Etiquette over the years. Ten save us, It is no exaggeration to say the very house might collapse on our heads without his work."_ 

## About The Pull Request
This PR intends to introduce some QOL features for royal servants in order to enhance their play, while introducing new aspects to the role such as the Butler / Housekeeper becoming the "Senior" and most trusted / honored house servant to the royal family, the leader of subordinate servants.

Primary features:

- [X] The Butler or Housekeeper role gains mildly enhanced skills to current, along with a medium wealthy pouch of money to start. (Sneaking about for evesdropping or stitching up wounds of princelings)

- [X]  Butler and Housekeeper given a random chance for music skills to entertain the royal family or other court nobles.

- [X] This role now requires characters of middle age or higher to play. Reflecting a trusted royal servant at the pinnacle of their career and primary keeper of the royal household. 

- [X]  Limits Butlers and Housekeepers to "Trusted" player species. (Human, Elf, Half-Elf, Dwarf, Aasimar)

- [X]  (WIP) A special Butler key is introduced, for (perhaps) a special wine reserve or (perhaps) pantry door to keep out pesky garrison from eating the Royal family's meals. Servant key ring is removed, Jester ring has implemented for Jester

- [X]  Butler and Housekeeper need special drip - Adds outfits using the king's colors in their uniform as senior house servants. 

- [x]  Servant roles reintroduced to be lead by the Butler or Housekeeper, with skills similar to old maid/butler status and be the Senior housekeeper's underlings to manage (Skills subject to change)

## Why It's Good For The Game

Maid and Butler are currently rather underplayed, but important for nobles to get things done, send messages do errands, cook meals, serve food, etc, etc etc, and generally boss around so they can ACT noble.

Adds another non-combat towner class to the mix with decent / appealing skills and roleplay to do.

This adds some depth to the role, rewarding a single dedicated maid or butler with status above the rest, with enhanced skills and money to reward playing, along with a bit of authority to be the "Head of the servants"

## **_WHY IT'S A DRAFT?_**

- Feedback, new ideas to play with the idea of a Head Butler / Majordomo / Seneschal 

- Better outfit ideas to distinguish the Housekeeper / Butler are welcome, right now their 

- GLOB.lordcolor feature for their clothes is entirely copypasted, feels sloppy, is sloppy.

- Waiting for map PR's to merge or close before adding any key requirements to the manor

- Servants got nuked with "Young adults" a few months back and re-adding them has been a PITA, figuring that part out.

## Pre-Merge Checklist
- [X] You tested this on a local server.
- [X] This code did not runtime during testing.
- [X] You documented all of your changes.

### CODE REVIEW

**Comment by ThePainkiller** [comment]
Not bad, not bad at all, good formatting even, though I'd love to see some screenshots of the king's colors and the butler/maid drip, but I like these

**Comment by Optimism333** [comment]
Looks interesting! A couple of suggestions.

- First, servants should have average knives. At least for now, we're going by the rule that pretty much all towners get average in ONE combat skill.
- Second, servant stat lineup should be something like the following:
+1 Endurance
+1 Speed
These are, in my opinion, the two most important stats for a role that needs to be on its feet and speedy, running errands from one end of the manor to the other.
- Perhaps Butler/Housekeeper stats can be something like the following:
+2 Intelligence
+1 Endurance
+1 Speed
-1 Strength

**Comment by AceVX** [comment]
> Not bad, not bad at all, good formatting even, though I'd love to see some screenshots of the king's colors and the butler/maid drip, but I like these

AXPliMt

Nothing special yet, Just a use of the same code as other lordcolor clothes on the current outfits.

I did give the butler noble high black boots, and in my dreamworld a Periwig and much fancier vest. But I'm not a sprite artist.

Another dreamworld would be to have the SECONDARY lordcolor be used but I'm testing further to figure out how that might work on the initialize using the copy pasted code.

**Comment by AceVX** [comment]
Gives Servants and Butlers Cardio stats for running errands and scrolls of their masters.

**Comment by lavillastrangiato** [comment]
So basically there would be a readdition of a "servant" role that would have the "adult" age, as well?

**Comment by AceVX** [comment]
> So basically there would be a readdition of a "servant" role that would have the "adult" age, as well?

Correct. The Servant role on this has been re-added with some modern SK updates.

File: code/modules/jobs/job_types/roguetown/peasants/butler.dm [hunk 1]
```diff
@@ -3,54 +3,58 @@
...
 		"Elf",
 		"Half-Elf",
 		"Dwarf",
-		"Tiefling",
-		"Dark Elf",
 		"Aasimar"
 	)
-	tutorial = "Servitude unto death, your blade a charcuterie of artisanal cheeses and meat, your armor wit and classical training. You dont consider yourself a slave, if anything youre a part of the family now. You alone have raised Kings, Queens, Princesses and Princees; without you the royals would have all starved to death."
+	allowed_ages = list(AGE_MIDDLEAGED, AGE_OLD, AGE_IMMORTAL)
+	tutorial = "Servitude unto death, your blade a charcuterie of artisanal cheeses and meat, your armor wit and classical training. Youre a part of the family now, the head of the Royal household staff and wear their colors. You alone have raised Kings, Queens, Princesses and Princees; without you the royals would have all starved to death."
```
Comment (BadAtThisGame302) [inline code/modules/jobs/job_types/roguetown/peasants/butler.dm:18]: Princes* not Princees.

File: code/modules/jobs/job_types/roguetown/peasants/butler.dm [hunk 2]
```diff
@@ -3,54 +3,58 @@
...
-	f_title = "Maid"
+	f_title = "Housekeeper"
 	allowed_races = list(
 		"Humen",
 		"Elf",
 		"Half-Elf",
 		"Dwarf",
-		"Tiefling",
-		"Dark Elf",
 		"Aasimar"
```
Comment (BadAtThisGame302) [inline code/modules/jobs/job_types/roguetown/peasants/butler.dm:15]: I personally think people wouldn't hire Aasimar to be the **head** of the household servantry, feels like another situation where their bruteforceness or maybe even lack of etiquette and such may conflict with their duties... or maybe I'm tripping idk Aasimar players don't shit on me too hard.

Comment (Optimism333) [reply code/modules/jobs/job_types/roguetown/peasants/butler.dm:15]: I think that's fair, tbh. An aasimar being used to labor as a servant? Sure. To manage servants? Nah.

File: code/modules/jobs/job_types/roguetown/peasants/butler.dm (hunk 1 above)
Comment (ThePainkiller) [reply code/modules/jobs/job_types/roguetown/peasants/butler.dm:18]: Also *you're

**Comment by AceVX** [comment]
image

Drip is good enough for now without further additions. Next step is implementing the Butler Key's uses in the pantry or wine cellar, private room in the basement. Shifting attention to Servants now.

**Comment by Optimism333** [comment]
I personally don't like giving the servants a -1 to strength. I think they would sit just fine at a +1 to Speed, and a +1 to Endurance, with no other significant maluses or bonuses.

Any reason in particular you think they should have that?

**Comment by AceVX** [comment]
> I personally don't like giving the servants a -1 to strength. I think they would sit just fine at a +1 to Speed, and a +1 to Endurance, with no other significant maluses or bonuses.
> 
> Any reason in particular you think they should have that?

This is a direct pull from present butler/maid statistics and confering it to the servants, Which are essentially taking their place as the mainstay grunt staff. Their stat adjusts lie at -1 str, +1 int, +1 perception. I think a counterbalance to a high speed stat is below average strength. There's no common use for it other than combat and a life of gentler labor wouldn't give them the strength we see in similar roles. Butcher for example gets +1 str / end and -1 int.

The goal of this is current maid/butler role but named 'servant' and there is be senior servant role named 'housekeeper/butler' which gets all of these reward buffs for being a "Professional" at this role. Much like we see the career steps elswhere. Squire, RG, Veteran, Captain etc.

**Comment by ThePainkiller** [comment]
The -1 STR also discourages them from becoming a combat class further, so I'm fine with the malus on that myself

**Comment by AceVX** [comment]
I think "As is" this RP accomplishes all its goals, with a second PR to come for map additions and uses for the new butler key. 

Servants take the place of the current butlers and maids with a few new tweaks, Role is implemented as I _basically_ envisioned it. Testing allowed me to select servant in both preferences and latejoin.

I'm not terribly happy with the Housekeeper's outfit but I restrained myself from giving her noble silks.

**Review by BadAtThisGame302 — APPROVED** [review]
Let us see how this plays out, especially with the different keys and such.

**Review by Optimism333 — APPROVED** [review]
My slight grumbling about the stats aside, this is fine. Looks ready to go to me.

**Review by Stutternov — APPROVED** [review]

**Pull request merged**